### ISSUE TITLE
Age Verify: That 32-bit datetime error

### ISSUE THREAD

**Issue by joelbrock** [CORE-POS/IS4C#840]
Received a report that age verification was causing a fatal crash:
`Feb 15 17:15:51 pos-lane corepos[3504]: (debug) DateTime::__construct(): Failed to parse time string (02261968) at position 7 (8): Unexpected character Line 557, C:\xampp\htdocs\Our-Table\pos\is4c-nf\lib\PrehLib.php`

Related to  https://github.com/CORE-POS/IS4C/commit/3708b0e45f7716d955c7f698b525548a39b45692

**Comment by gohanman** [comment]
I'm not sure what to make of that. It works with every version on PHP released this decade (https://3v4l.org/Xp05O) and fails differently if one insists on using something older than that.

**Comment by gohanman** [comment]
For the sake of clarity, what that link is demonstrating is that in PHP >= 5.2.0 this exception is caught as expected rather than causing a fatal error. Earlier versions crash with different errors because either DateTime doesn't exist or the syntax isn't valid in PHP4.

I don't think there's a bug here. My guess is the install in question is only caught up to f69bd76ff561ee2808c3543166e02ddd4f366c47 and doesn't have the try/catch.

**Comment by gohanman** [comment]
Never mind, there is a bug e8de5814651ccaddea5948f80d5a966628de5ae3

What's happening here is rather unexpected. Because of namespacing the code really **means** this:
```php
try {
    $dt = new DateTime('foo');
} catch (COREPOS\pos\lib\Exception $ex) {
}
```
Because `COREPOS\pos\lib\Exception` is not a class, the catch block doesn't match and the exception isn't caught. But no error is raised indicating `COREPOS\pos\lib\Exception` doesn't exist.

**Comment by gohanman** [comment]
Fixed in 2.5
